### PR TITLE
Drop return typehint on QuerySpecification

### PR DESCRIPTION
**Pull request by Brammm**
Addresses #12

### CODE REVIEW

File: tests/Stub/FieldEquals.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ public function __construct(string $field, $value)
         $this->value = $value;
     }
 
-    public function asQueryConstraint(QueryBuilder $queryBuilder, string $rootAlias): ?object
+    public function asQueryConstraint(QueryBuilder $queryBuilder, string $rootAlias)
```
Comment (frankdejonge) [inline tests/Stub/FieldEquals.php]: This change is probably not needed, right?

**Comment by frankdejonge** [comment]
@Brammm looks good! Can you add a testcase that validates support?

**Comment by Brammm** [comment]
@frankdejonge I just noticed how you circumvented the whole "`isNull`returns string" thing in the `FieldIsNotNull` test:

```php
return $expr->andX($expr->isNotNull("{$rootAlias}.{$this->field}"));
```

So I could either change that test to return the string (and make sure everything still works) _or_ drop this PR and do the same in my project...

**Comment by frankdejonge** [comment]
@Brammm both work for me :)

**Pull request closed without merging**